src/lib/utils/PushNotification.ts

```diff
@@ -1,7 +1,75 @@
 import AsyncStorage from "@react-native-community/async-storage"
-import PushNotification from "react-native-push-notification"
+import { navigate } from "lib/navigation/navigate"
+import { globalStoreInstance } from "lib/store/GlobalStore"
+import { PendingPushNotification } from "lib/store/PendingPushNotificationModel"
+import PushNotification, { ReceivedNotification } from "react-native-push-notification"
 import { ASYNC_STORAGE_PUSH_NOTIFICATIONS_KEY } from "./AdminMenu"
 
+const MAX_ELAPSED_TAPPED_NOTIFICATION_TIME = 90 // seconds
+
+export const handlePendingNotification = (notification: PendingPushNotification) => {
+  const elapsedTimeInSecs = Math.floor((Date.now() - notification.tappedAt) / 1000)
+  if (elapsedTimeInSecs <= MAX_ELAPSED_TAPPED_NOTIFICATION_TIME && !!notification.data.url) {
+    navigate(notification.data.url, { passProps: { ...notification.data, url: undefined } })
```

Comment:
No need to set `url` to undefined:

```suggestion
    navigate(notification.data.url, { passProps: { ...notification.data } })
```

Reply:
I wouldn't want to pass `url` as prop to any component unintentionally.

Reply:
Ah I see, so you intentionally unsetting it here, right? That's a pretty good idea 👍

Reply:
Thanks for the clarification :)

Reply:
good thinking but I don't think that would cause problems since I don't see [navigate](https://github.com/artsy/eigen/blob/ea6754df375ffe4d8cc180fb08781602ddba93d6/src/lib/navigation/navigate.ts) using it anywhere. Is it being used somewhere maybe where I am not aware of it?

Reply:
@MounirDhahri navigate will pass it as a prop to the next component. I was just afraid that it is easy for any Dev to just choose url as a prop name (without thinking twice) to pass whatever urls including external urls that might pop open webviews. If this notification lands on such, we'd have a very weird bug.

Reply:
Yes, I understand that's true but I don't believe it will be triggering opening webviews since the navigate is based on the first param (the URL param) and does not take in count merging both params (URL and passProps). This is not a blocker at all but It's something I don't see as necessary, we can of course leave it as it is with no issues